proto/api/v1/cloud_source_service.proto

```diff
@@ -0,0 +1,155 @@
+option java_package = "io.stackrox.proto.api.v1";
+
+import weak "google/api/annotations.proto";
+import "api/v1/empty.proto";
+import "api/v1/pagination.proto";
+import "gogoproto/gogo.proto";
+
+package v1;
+
+// CloudSource is an integration which provides a source for discovered
+// clusters.
+message CloudSource {
+    string id                            = 1;
+    string name                          = 2;
+    string type                          = 3;
+    bool   skip_test_integration         = 4;
+
+    oneof Config {
+        PaladinCloudConfig paladin_cloud = 5;
+        OCMConfig          ocm           = 6;
+    }
+}
+
+// PaladinCloudConfig provides information required to fetch discovered
+// clusters from Paladin Cloud.
+message PaladinCloudConfig {
+    string endpoint = 1 [(gogoproto.moretags) = 'scrub:"dependent" validate:"nolocalendpoint"'];
+    string token    = 2 [(gogoproto.moretags) = 'scrub:"always"'];
+}
+
+// OCMConfig provides information required to fetch discovered clusters from
+// the OpenShift cluster manager.
+message OCMConfig {
+    string endpoint = 1 [(gogoproto.moretags) = 'scrub:"dependent" validate:"nolocalendpoint"'];
+    string token    = 2 [(gogoproto.moretags) = 'scrub:"always"'];
+}
+
+message CloudSourcesFilter {
+    // Matches cloud sources based on their type.
+    repeated string types = 1;
+}
+
+message CountCloudSourcesRequest {
+    // Filters cloud sources based on the requested fields.
+    CloudSourcesFilter filter = 1;
+}
+
+message CountCloudSourcesResponse {
+    int32 count = 1;
+}
+
+message GetCloudSourceRequest {
+    string id = 1;
+}
+
+message GetCloudSourceResponse {
+    CloudSource item = 1;
+}
+
+message ListCloudSourcesRequest {
+    // Divides the response into chunks.
+    Pagination pagination = 1;
+
+    // Filters cloud sources based on the requested fields.
+    CloudSourcesFilter filter = 2;
+}
+
+message ListCloudSourcesResponse {
+    repeated CloudSource items = 1;
+}
+
+message PostCloudSourceRequest {
+    CloudSource item = 1;
+}
+
+message PostCloudSourceResponse {
+    CloudSource item = 1;
+}
+
+message PutCloudSourceRequest {
+    CloudSource item        = 1;
+    // If true, item must include valid credentials.
+    // If false, the resource must already exist and
+    // credentials in item are ignored.
+    bool update_credentials = 2;
+}
+
+message DeleteCloudSourceRequest {
+    string id = 1;
+}
+
+message TestCloudSourceRequest {
+    CloudSource item        = 1;
+    // If true, item must include valid credentials.
+    // If false, the resource must already exist and
+    // credentials in item are ignored.
+    bool update_credentials = 2;
+}
+
+service CloudSourcesService {
+    // CountCloudSources returns the number of cloud sources after filtering by requested fields.
+    rpc CountCloudSources(CountCloudSourcesRequest) returns (CountCloudSourcesResponse) {
+        option (google.api.http) = {
+            get: "/v1/count/cloudsources"
```

Comment:
general question: how do we feel about `/v1/cloud-sources`, i.e. using kebab case? Personally like it more for APIs consisting of two or multiple words (as said, personal pref, since we don't really have guidelines for that at the moment).

Reply:
I also like the hyphenated URLs more, but I chose non-hyphenated because that's what we have in other integrations (e.g. `/v1/imageintegrations`, `/v1/externalbackups`). That said there are already some changes with the new guidelines, I think we would have the liberty to hyphenate if we prefer.